### DISCUSSION TITLE
Have you encountered system hang or IO device error? Here's my workaround

### DISCUSSION THREAD

**Discussion by superpi15** [microsoft/WSL#7869] [General]
Environment: 
- Host: Windows 11 Pro
- WSL Ubuntu 20.04
- Working directory: a vhdx located external M2 SSD disk connect by USB mounted via wsl command

My case: 
- Whenever I run large software such as ModelSim, my WSL stuck. (tried several times)
- Besides, the installation of the latest Xilinx toolchain (2021.2), which requires hundreds giga bytes, also hang during installation. (tried several times)
- You will see the disk stays 100% active but 0 data transfer. Sometime it tells you IO device error. 

Guess: Maybe it is about external disk right? The hang may be also related to filesystem. 

My workaround solution: 
Simplify your environmental PATH when you work with external disk. Specifically, export PATH in such away, "NO 9P FILESYSTEM REGION." You remove any searching path directly to Windows directory outside your vhdx. 
Then my ModelSim and Xilinx toolchain installation work perfectly without any problem.

**Comment by rvanlaak** [comment]
Would this be related to issues like https://github.com/microsoft/WSL/issues/7754 and https://github.com/microsoft/WSL/issues/7883 and https://github.com/microsoft/WSL/issues/7782

**Reply by superpi15** [reply]
idk bro. 
Thank you for the links anyway.

**Reply by rvanlaak** [reply]
It seems Microsoft is starting to look into it: https://github.com/microsoft/WSL/issues/7883#issuecomment-1005986074

**Comment by rvanlaak** [comment]
Another report of Windows hanging https://github.com/microsoft/WSL/issues/7945
